internal/services/access_targets.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/apono-io/terraform-provider-apono/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"golang.org/x/exp/slices"
+)
+
+func ConvertIntegrationTargetsApiToTerraformModel(ctx context.Context, aponoClient *apono.APIClient, integrationTargets []apono.AccessTargetIntegrationV1) ([]models.IntegrationTarget, diag.Diagnostics) {
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var dataIntegrationTargets []models.IntegrationTarget
+	for _, integrationTarget := range integrationTargets {
+		integration, diagnostics := convertIntegrationTargetApiToTerraformModel(ctx, &integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataIntegrationTargets = append(dataIntegrationTargets, *integration)
+	}
+
+	return dataIntegrationTargets, nil
+}
+
+func ConvertIntegrationTargetsTerraformModelToApi(ctx context.Context, aponoClient *apono.APIClient, integrationTargets []models.IntegrationTarget) ([]apono.AccessTargetIntegrationV1, diag.Diagnostics) {
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var resultIntegrationTargets []apono.AccessTargetIntegrationV1
+	for _, integrationTarget := range integrationTargets {
+		integration, diagnostics := convertIntegrationTargetTerraformModelToApi(integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		resultIntegrationTargets = append(resultIntegrationTargets, *integration)
+	}
+
+	return resultIntegrationTargets, nil
+}
+
+func ConvertBundleTargetsApiToTerraformModel(ctx context.Context, aponoClient *apono.APIClient, bundleTargets []apono.AccessTargetBundleV1) ([]models.BundleTarget, diag.Diagnostics) {
+	availableBundles, _, err := aponoClient.AccessBundlesApi.ListAccessBundles(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "access bundles", "")
+	}
+
+	var dataBundleTargets []models.BundleTarget
+	for _, bundleTarget := range bundleTargets {
+		bundle, diagnostics := convertBundleTargetApiToTerraformModel(&bundleTarget, availableBundles.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataBundleTargets = append(dataBundleTargets, *bundle)
+	}
+
+	return dataBundleTargets, nil
+}
+
+func ConvertBundleTargetsTerraformModelToApi(ctx context.Context, aponoClient *apono.APIClient, bundleTargets []models.BundleTarget) ([]apono.AccessTargetBundleV1, diag.Diagnostics) {
+	availableBundles, _, err := aponoClient.AccessBundlesApi.ListAccessBundles(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "bundles", "")
```

Comment:
why here it's "list" "bundles" and other place (line 55) its "list" "access bundles"

Reply:
mistake, will change to access bundles